### PR TITLE
5.7 PS-5642 - page tracker thread won't exist if startup fails

### PR DESCRIPTION
**Pull request by altmannmarcelo**
If page tracker is enabled and innodb fail to start,
shutdown process won't signal page tracker thread to exit.

This fix ensure those threads will be proper signaled and exit.

### CODE REVIEW

**Comment by altmannmarcelo** [comment]
https://ps57.cd.percona.com/job/percona-server-5.7-param/145/

File: mysql-test/suite/innodb/t/percona_changed_page_bmp_shutdown_thread.test [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+--error 1
+--exec $MYSQLD_CMD --innodb_track_changed_pages=TRUE --debug=d,ib_dic_boot_error --log-error=$MYSQLTEST_VARDIR/tmp/percona_changed_page_bmp_shutdown_thread.err
+
+--echo # Asserting that the error was not thrown.
+--let log_file_full_path=$MYSQLTEST_VARDIR/tmp/percona_changed_page_bmp_shutdown_thread.err
+--let grep_pattern=threads created by InnoDB had not exited at shutdown
+--let log_expected_matches=0
+--source include/log_grep.inc
+
+--echo # Asserting that the server aborted.
```
Comment (laurynas-biveinis) [inline mysql-test/suite/innodb/t/percona_changed_page_bmp_shutdown_thread.test]: Same comment as for the 8.0 PR

**Comment by altmannmarcelo** [comment]
ready for next review

**Review by dutow — CHANGES_REQUESTED** [review]
Same comments as for the 5.6 version

**Comment by altmannmarcelo** [comment]
@dutow ready for next review.

**Comment by altmannmarcelo** [comment]
@dutow ready for next review

File: storage/innobase/srv/srv0start.cc [hunk 2]
```diff
@@ -1451,6 +1451,12 @@ srv_shutdown_all_bg_threads()
 			if (srv_n_fil_crypt_threads_started) {
 				os_event_set(fil_crypt_threads_event);
 			}
+
+			/* Stop srv_redo_log_follow_thread thread */
+			if (srv_redo_log_thread_started) {
```
Comment (dutow) [inline storage/innobase/srv/srv0start.cc:1456]: Is 5.7 affected then? If yes, the jira ticket should also be updated.

**Comment by altmannmarcelo** [comment]
@dutow fixed. Added 5.6 and 5.7 as affected.

**Review by dutow — APPROVED** [review]
5.6 isn't affected, you only ported the testcase there. Other than that looks good.

**Pull request merged**